docs-next/pages/guides/cli.mdx

```diff
@@ -2,81 +2,188 @@ import { Markdown } from '../../components/Page';
 
-#### keystone-next generate
+While Keystone abstracts much of the Prisma workflow for you, we strongly recommend you familiarise yourself with the Prisma Migrate CLI commands when you are managing your application in production.
 
-> keystone-next generate is only usable when db.useMigrations is set to true in your config
+## Bringing it all together
 
-This command will generate a migration based on the current state of your database and your Keystone schema.
-The generated migration artefact should be added to your repository so that it can be shared with other developers and deployed in production.
-You should generally use `keystone-next dev` instead of `keystone-next generate` because it will prompt you to create a migration when you need to.
-You only need to use `keystone-next generate` when you want to generate an empty migration to do something that isn't a result of a schema change or generate a migration in a non-interactive environment.
-`keystone-next generate` accepts the following options as command line arguments or as prompts:
+Although there are many subtle ways you might customise your workflow with Keystone based on your project's requirements, this is a fairly typical way of working:
 
-- `--name` to set the name of the migration
-- `--accept-data-loss` to allow resetting the database when it is out of sync with the migrations
-- `--allow-empty` to create an empty migration when there are no changes to the schema
+### Postinstall after git clone and pull
 
-#### keystone-next deploy
+Setting up the `postinstall` script will make sure that Keystone's generated files are all present and correct when you get latest from source control and install dependencies.
+
+This will also run in CI, which means your lint rules and tests will be using the latest version of your Keystone build.
+
+### Developing your project
+
+Use `keystone-next dev` to start Keystone for development. Make sure you commit any changes to generated files, including both the Prisma and GraphQL Schema files, and migrations.
+
+### Deplying to production
+
+Most application hosting platforms allow you to specify a `build` script (for new deployments) and a `start` script (for when the app is started, including new instances when scaling up).
+
+#### Build Script
+
+Install the project dependencies, generate the build and deploy migrations:
+
+```bash
+yarn && yarn keystone-next build && yarn keystone-next prisma migrate deploy
+```
+
+x> Note: it is only safe to run migrations in the build step if deploys are built synchronously by your hosting provider. Make sure you don't run migrations against your production database from staging / preview builds.
```

Comment:
"built synchronously" doesn't make sense to me.

Reply:
It means the hosting provider doesn't run multiple builds in parallel for different deploys

That could get ... weird

Can you think of a better way to word it?

Reply:
> run multiple builds in parallel for different deploys

Still not totally sure what you mean. Could you describe the problematic scenario?